### ISSUE TITLE
StereoAudioRecorder doesn't actually fully disconnect

### ISSUE THREAD

**Issue by philazzi44** [muaz-khan/RecordRTC#24]
While leveraging RecordRTC in Chrome, I stumbled on the fact that the script node does not actually stop firing when "stopRecording()" is fired on the RecordRTC instance.

I'm not sure if this is the correct fix, but testing locally revealed that modifying the StereoAudioRecorder's stop function to also disconnect the __stereoAudioRecorderJavacriptNode instance resolved the problem.

``` javascript
this.stop = function(callback) {
...
    // to make sure onaudioprocess stops firing
    audioInput.disconnect();
    __stereoAudioRecorderJavacriptNode.disconnect(); // New line
...
}
```

Basically, without this, it seems that the __stereoAudioRecorderJavacriptNode.onaudioprocess keeps firing, with the "recording" flag set to true, resulting in the following being called over and over, without end.

``` javascript
__stereoAudioRecorderJavacriptNode.onaudioprocess = function (e) {
...
    if (!recording) {
        audioInput.disconnect();
        return;
    }
...
}
```

Also, when performing multiple recordings, some recordings ended up with a blob that only contained the 44 header bytes (and no data). Haven't been able to prove it, but my running theory is that due to having multiple script nodes actively firing, the data was never processed properly by the desired script node. Haven't been able to reproduce this after introducing this fix locally.

**Comment by mreinsch** [comment]
I'm also seeing some issue that could be related to this one: the first recording is fine, but after that each  recording is reported as "audio/wav -> 44.0 Bytes". Playback then also fails of course, putting the data URI into jPlayer results in an error.

That was working properly with the version from Feb 12, 2015, 08:32:23. Haven't tested any versions in-between. I'll try applying the described fix, and see how it goes.

**Comment by mreinsch** [comment]
hm, this doesn't seem to fix it for me

**Comment by philazzi44** [comment]
I'm assuming this is a recording done in Chrome? Your described issue does sound familiar, and the fix I described was to account for the multiple scripts running causing some kind of contention. I found it helpful to dump a bunch of temporary logging in some of the RecordRTC inner classes, specifically the ones that do the audio/video processing to see whether they're firing, then identify why they were (or weren't) firing when I expected.

I haven't updated our codebase to a newer version of the entire RecordRTC because there were too many changes, so the version we're using successfully is from around April 2015.
